### PR TITLE
WIP: Pep8 and comments

### PR DESCRIPTION
**Pull request by dneise**
commenting some functions as I understand them and pep8

### CODE REVIEW

**Comment by dneise** [comment]
found a bug in `pulse_shape.py`

Added an image to README

File: digicamtoy/tracegenerator.py [hunk 1]
```diff
@@ -36,13 +36,56 @@ def _generalized_poisson_crosstalk(crosstalk, n_photon):
...
+        baseline=200,
+        time_signal=20,
+        jitter=0,
+        pulse_shape_file='/utils/pulse_SST-1M_pixel_0.dat',
+        sub_binning=0,
+        **kwargs
+    ):
+        '''
+        time_start: scalar
+            start time of sampled waveforms (in ns?).
```
Comment (calispac) [inline digicamtoy/tracegenerator.py]: yes in ns

File: digicamtoy/tracegenerator.py [hunk 2]
```diff
@@ -36,13 +36,56 @@ def _generalized_poisson_crosstalk(crosstalk, n_photon):
...
+
+        time_end: scalar
+            end time of sampled waveforms in the same unit as time_start.
+
+        time_sampling: scalar
+            number of samples per unit of time. or units of time per sample.
```
Comment (calispac) [inline digicamtoy/tracegenerator.py]: width of the time sampling e.g. time_sampling = 4 ns for DigiCam

File: digicamtoy/tracegenerator.py [hunk 3]
```diff
@@ -36,13 +36,56 @@ def _generalized_poisson_crosstalk(crosstalk, n_photon):
...
+            (digicam samples with 250MHz so a sample is 4ns long.)
+
+        sigma_e: scalar or 1-d array of shape (n_pixels, )
+            electronics noise (std-dev of normal distribution)
+            electronics noise is simulated as white noise.
+
+        jitter: scalar or 1d or 2d array of shape: ???
```
Comment (calispac) [inline digicamtoy/tracegenerator.py]: jitter is the time jitter for photon arrival time (only signal photons)

File: digicamtoy/tracegenerator.py [hunk 4]
```diff
@@ -36,13 +36,56 @@ def _generalized_poisson_crosstalk(crosstalk, n_photon):
...
+            I do not know what kind of jitter is meant here, and in what
+            unit it might be given.
+
+        n_photon: sclar or 1d or 2d of shape: ???
```
Comment (calispac) [inline digicamtoy/tracegenerator.py]: scalar would create same number of photons for all pixels. (signal photons affected by jitter)
1d with shape =(n_pixels,) would put each photon in each pixel
2d with shape = (n_pixels, ?) would be a list of photons per pixel

File: digicamtoy/tracegenerator.py [hunk 5]
```diff
@@ -36,13 +36,56 @@ def _generalized_poisson_crosstalk(crosstalk, n_photon):
...
+            the expectation value of the number of photons
+            per pixel and per event.
+            The actual number of photons per pixel and per event is drawn from
+            a poisson distribution, with the expectation value `n_photons`.
+            Even negative values are allowed.
```
Comment (calispac) [inline digicamtoy/tracegenerator.py]: negative would return 0 photons I think. maybe we can warn for this

File: digicamtoy/tracegenerator.py [hunk 6]
```diff
@@ -36,13 +36,56 @@ def _generalized_poisson_crosstalk(crosstalk, n_photon):
...
+
+        time_signal: scalar or 1d or 2d of shape: ???
```
Comment (calispac) [inline digicamtoy/tracegenerator.py]: shape same as for n_photons

**Comment by calispac** [comment]
> found a bug in pulse_shape.py

One was used before camera assembly (*after_preamp*.txt). It is as is suggested after the preamp (before DigiCam).

The other is taken from camera commissioning (only for pixel_id = 0) after DigiCam. 

We should use the second!

File: digicamtoy/tracegenerator.py (hunk 3 above)
Comment (dneise) [reply digicamtoy/tracegenerator.py]: in ns?

**Comment by dneise** [comment]
> We should use the second!

We should make this very clear somehow... Maybe event removing the one which is not useful in the toy monte carlo?

File: digicamtoy/tracegenerator.py (hunk 3 above)
Comment (calispac) [reply digicamtoy/tracegenerator.py]: yes ns. we can add that everything time related is ns

**Comment by calispac** [comment]
Well default template is the second one (default value at initialization). I don't think we should remove it maybe we want to go to previous results

**Pull request closed without merging**